sdk/eventhub/event-hubs/src/eventProcessor.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { EventHubClient } from "./eventHubClient";
+import { EventPosition } from "./eventPosition";
+import { PartitionContext } from "./partitionContext";
+import { CheckpointManager, Checkpoint } from "./checkpointManager";
+import { EventData } from "./eventData";
+
+export interface PartitionProcessor {
+  /**
+   * Optional. Called when EPH begins processing a partition.
+   */
+  initialize?(): Promise<void>;
+  /**
+   * Optional. Called when EPH stops processing a partition.
+   * This may occur when control of the partition switches to another EPH or when user stops EPH
+   * TODO: update string -> CloseReason
+   */
+  close?(reason: string): Promise<void>;
+  /**
+   * Called when a batch of events have been received.
+   */
+  processEvents(events: EventData[]): Promise<void>;
+  /**
+   * Called when the underlying client experiences an error while receiving.
+   */
+  processError(error: Error): Promise<void>;
+}
+
+/**
+ * used by PartitionManager to claim ownership.
+ * returned by listOwnerships
+ */
+export interface PartitionOwnership {
+  eventHubName: string;
+  consumerGroupName: string;
+  instanceId: string;
+  partitionId: string;
+  ownerLevel: number;
+  offset?: number;
+  sequenceNumber?: number;
+  lastModifiedTime?: number;
+  ETag?: string;
+}
+
+/**
+ * The PartitionProcessorFactory is called by EPH whenever a new partition is about to be processed.
+ */
+export interface PartitionProcessorFactory {
+  createPartitionProcessor(
```

Comment:
For JavaScript, it's more idiomatic to provide a function as the factory than a class/object that has a factory method on it.

If you remove the function name `createPartitionProcessor` but leave the rest of the signature, then `PartitionProcessorFactory` will just describe a function instead.

So this should work:
```typescript
const partitionProcessorFactory: PartitionProcessorFactory = (
  context: PartitionContext,
  checkpointManager: CheckpointManager
) => {
  return new PartitionProcessor();
};
```

This was discussed in the API proposal but we had the existing example to make it easier for our C# friends to read :)

Reply:
Nice!!